.gitlab-ci.yml

```diff
@@ -696,149 +696,84 @@ build_dogstatsd:
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/docker-machine:latest
   services:
     - 486234852809.dkr.ecr.us-east-1.amazonaws.com/docker-machine:latest
-  before_script: [ "# noop" ] # Override top level entry
-  dependencies: [] # Don't download Gitlab artefacts
-  script:
+  before_script:
+    - export SRC_TAG=v$CI_PIPELINE_ID-${CI_COMMIT_SHA:0:7}
     - export DOCKER_HOST=tcp://486234852809.dkr.ecr.us-east-1.amazonaws.com__docker-machine:2375
     - eval "$(aws ecr get-login --region us-east-1 --no-include-email --registry-ids 486234852809)"
     - DOCKER_REGISTRY_LOGIN=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.$DOCKER_REGISTRY_LOGIN_SSM_KEY --with-decryption --query "Parameter.Value" --out text)
     - aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.$DOCKER_REGISTRY_PWD_SSM_KEY --with-decryption --query "Parameter.Value" --out text | docker login --username "$DOCKER_REGISTRY_LOGIN" --password-stdin "$DOCKER_REGISTRY_URL"
-    - TAG_SUFFIX=${TAG_SUFFIX:-}
-    - LATEST_TAG=${LATEST_TAG:-latest}
-    - docker pull $SOURCE_IMAGE:v$CI_PIPELINE_ID-${CI_COMMIT_SHA:0:7}$TAG_SUFFIX
-    - docker tag $SOURCE_IMAGE:v$CI_PIPELINE_ID-${CI_COMMIT_SHA:0:7}$TAG_SUFFIX $DEST_IMAGE:$LATEST_TAG$TAG_SUFFIX
-    - docker push $DEST_IMAGE:$LATEST_TAG$TAG_SUFFIX
+    - pip install invoke
+  dependencies: [] # Don't download Gitlab artefacts
 
 .docker_hub_variables: &docker_hub_variables
   DOCKER_REGISTRY_LOGIN_SSM_KEY: docker_hub_login
   DOCKER_REGISTRY_PWD_SSM_KEY: docker_hub_pwd
   DOCKER_REGISTRY_URL: docker.io
+  SRC_AGENT: *agent_ecr
+  SRC_DSD: *dogstatsd_ecr
+  SRC_DCA: *cluster-agent_ecr
 
 .quay_variables: &quay_variables
+  <<: *docker_hub_variables
   DOCKER_REGISTRY_LOGIN_SSM_KEY: quay_login
   DOCKER_REGISTRY_PWD_SSM_KEY: quay_pwd
   DOCKER_REGISTRY_URL: quay.io
 
-agent6_dev_docker_hub:
-  <<: *docker_tag_job_definition
-  when: manual
-  except:
-    - master
-  variables:
-    <<: *docker_hub_variables
-    SOURCE_IMAGE: *agent_ecr
-    DEST_IMAGE: datadog/agent-dev
-    LATEST_TAG: $CI_COMMIT_REF_SLUG
-
-agent6_jmx_dev_docker_hub:
-  <<: *docker_tag_job_definition
-  when: manual
-  except:
-    - master
-  variables:
-    <<: *docker_hub_variables
-    SOURCE_IMAGE: *agent_ecr
-    DEST_IMAGE: datadog/agent-dev
-    LATEST_TAG: $CI_COMMIT_REF_SLUG
-    TAG_SUFFIX: -jmx
-
-agent6_dev_docker_hub_master:
-  <<: *docker_tag_job_definition
-  only:
-    - master
-  variables:
-    <<: *docker_hub_variables
-    SOURCE_IMAGE: *agent_ecr
-    DEST_IMAGE: datadog/agent-dev
-    LATEST_TAG: $CI_COMMIT_REF_SLUG
-
-agent6_jmx_dev_docker_hub_master:
-  <<: *docker_tag_job_definition
-  only:
-    - master
-  variables:
-    <<: *docker_hub_variables
-    SOURCE_IMAGE: *agent_ecr
-    DEST_IMAGE: datadog/agent-dev
-    LATEST_TAG: $CI_COMMIT_REF_SLUG
-    TAG_SUFFIX: -jmx
-
-cluster_agent_dev_docker_hub:
+dev_branch_docker_hub:
```

Comment:
WDYT about renaming this to `agent_dev_branch_docker_hub` this would avoid `dca_dev_branch_docker_hub` being at the top in the pipeline
![screen shot 2018-06-22 at 10 36 24](https://user-images.githubusercontent.com/2368736/41766809-49b148cc-7608-11e8-9425-8fbece4344bc.png)

Reply:
Indeed, the ordering is not ideal. But as the step also pushes dogstatsd, I wanted to avoid calling it agent.
Our image steps are currently split between the `image_deploy` and `deploy` stages. WDYT about moving all the agent&dsd ones to `deploy`, and create a `cluster-agent-deploy` stage for the dca steps?